src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item/ProteinsListItem.tsx

```diff
@@ -39,13 +43,47 @@ type Props = {
 const ProteinsListItem = (props: Props) => {
   const { transcript, trackLength } = props;
 
-  const toggleListItemInfo = () => props.toggleExpandedProtein(transcript.id);
+  const dispatch = useDispatch();
+  const params: { [key: string]: string } = useParams();
+  const { genomeId, entityId } = params;
+  const { search } = useLocation();
+  const transcriptIdToFocus = new URLSearchParams(search).get('transcriptId');
+
+  const toggleListItemInfo = () => {
+    if (transcriptIdToFocus) {
+      const url = urlFor.entityViewer({
+        genomeId,
+        entityId,
+        view: View.PROTEIN
+      });
+      dispatch(replace(url));
+    }
+    props.toggleExpandedProtein(transcript.id);
+  };
 
   const midStyles = classNames(transcriptsListStyles.middle, styles.middle);
 
+  const itemRef = useRef<HTMLDivElement | null>(null);
+  useEffect(() => {
+    if (transcript.id === transcriptIdToFocus) {
+      // give it some time for other expanded proteiens to load
+      setTimeout(() => {
```

Comment:
I tried running it without the timeout, and it worked fine. At the point when `ProteinsListItem` has mounted, other `ProteinsListItem`s should exist as well. Is the timeout really necessary for anything?

Reply:
This was causing some issue when the newly focussed protein needs to expand along with the proteins that were already expanded previously. Now that we expand only the focussed protein, it will not be an issue anymore. I'll get it removed.